pkg/aggregator/aggregator.go

```diff
@@ -877,3 +879,41 @@ func (agg *BufferedAggregator) updateChecksTelemetry() {
 	}
 	t.Flush()
 }
+
+// deregisterSampler is an item sent internally by the aggregator to
+// signal that the sender will no longer will be used for a given
+// check.ID.
+//
+// 1. A check is unscheduled while running.
+//
+// 2. Check sends metrics, sketches and eventually a commit, using the
+// sender to the senderItems channel.
+//
+// 3. Check collector calls deregisterSampler() immediately as check's
+// Run() completes.
+//
+// 4. Metrics are still in the channel, we can't drop the
+// sampler. Instead we add another message to the channel (this type)
+// to indicate that the sampler will no longer be in use once all the
+// metrics are handled.
+//
+// 5. Aggregator handles metrics and the commit message. Sampler now
+// contains metrics from the check.
+//
+// 6. Aggragator handles this message. We can't drop the sender now
+// since it still contains metrics from the last run. Instead, we set
+// a flag that the sampler can be dropped after the next flush.
+//
+// 7. Aggregator flushes metrics from the sampler and can now remove
+// it: we know for sure that no more metrics will arrive.
+type deregisterSampler struct {
+	id check.ID
+}
+
+func (s *deregisterSampler) handle(agg *BufferedAggregator) {
+	agg.mu.Lock()
+	defer agg.mu.Unlock()
+	if cs, ok := agg.checkSamplers[s.id]; ok {
+		cs.deregistered = true
+	}
```

Comment:
Can you create a function for these lines in `BufferedAggregator` in order to encapsulate this logic inside `*BufferedAggregator`?